src/clients/AcrossConfigStoreClient.ts

```diff
@@ -385,23 +392,16 @@ export class AcrossConfigStoreClient {
           continue;
         }
 
-        // Extract last version
-        const lastValue =
-          this.cumulativeConfigStoreVersionUpdates.length === 0
-            ? DEFAULT_CONFIG_STORE_VERSION
-            : Number(
-                this.cumulativeConfigStoreVersionUpdates[this.cumulativeConfigStoreVersionUpdates.length - 1].value
-              );
-
-        // If version is not > last version, skip.
+        // Extract the current highest version. Require that the version always increments, otherwise skip the update.
+        const lastValue = Number(this.cumulativeConfigStoreVersionUpdates[0]?.value ?? DEFAULT_CONFIG_STORE_VERSION);
         if (value <= lastValue) {
           continue;
         }
 
-        this.cumulativeConfigStoreVersionUpdates.push({
-          ...args,
-          timestamp: globalConfigUpdateTimes[i],
-        });
+        // Prepend the update to impose descending ordering for version updates.
+        this.cumulativeConfigStoreVersionUpdates = [{ ...args, timestamp: globalConfigUpdateTimes[i] }].concat(
```

Comment:
I think we've been using the spread operator `[...a, ...b]` instead of `concat` - should we retain that usage?

Reply:
The differences between `.concat()` and the spread operator otherwise don't seem to be very important for our use cases, so I'd probably prefer to use the one that looks the least ugly. Prettier was actually butchering the code with `.concat()` and the spread operator avoids that, so that's an improvement!